src/pages/OnboardingAccounting/BaseOnboardingAccounting.tsx

```diff
@@ -89,177 +141,208 @@ function BaseOnboardingAccounting({shouldUseNativeStyles}: BaseOnboardingAccount
-                        adminsChatReportID,
-                        onboardingPolicyID: policyID,
-                        companySize: onboardingCompanySize,
-                        userReportedIntegration,
-                    });
+        const shouldCreateWorkspace = !onboardingPolicyID && !paidGroupPolicy;
 
-                    if (onboardingCompanySize !== CONST.ONBOARDING_COMPANY_SIZE.MICRO && getPlatform() !== CONST.PLATFORM.DESKTOP) {
-                        if (CONFIG.IS_HYBRID_APP) {
-                            return;
-                        }
-                        openOldDotLink(CONST.OLDDOT_URLS.INBOX, true);
-                    }
-                    // Avoid creating new WS because onboardingPolicyID is cleared before unmounting
-                    InteractionManager.runAfterInteractions(() => {
-                        setOnboardingAdminsChatReportID();
-                        setOnboardingPolicyID();
-                    });
+        // We need `adminsChatReportID` for `completeOnboarding`, but at the same time, we don't want to call `createWorkspace` more than once.
+        // If we have already created a workspace, we want to reuse the `onboardingAdminsChatReportID` and `onboardingPolicyID`.
+        const {adminsChatReportID, policyID} = shouldCreateWorkspace
+            ? createWorkspace(undefined, true, '', generatePolicyID(), CONST.ONBOARDING_CHOICES.MANAGE_TEAM, '', undefined, false, onboardingCompanySize)
+            : {adminsChatReportID: onboardingAdminsChatReportID, policyID: onboardingPolicyID};
 
-                    // We need to wait the policy is created before navigating out the onboarding flow
-                    Navigation.setNavigationActionToMicrotaskQueue(() => {
-                        navigateAfterOnboarding(
-                            onboardingPurposeSelected,
-                            isSmallScreenWidth,
-                            canUseDefaultRooms,
-                            policyID,
-                            activeWorkspaceID,
-                            adminsChatReportID,
-                            // Onboarding tasks would show in Concierge instead of admins room for testing accounts, we should open where onboarding tasks are located
-                            // See https://github.com/Expensify/App/issues/57167 for more details
-                            (session?.email ?? '').includes('+'),
-                        );
-                    });
-                }}
-                isLoading={isLoading}
-                isDisabled={isOffline && onboardingCompanySize !== CONST.ONBOARDING_COMPANY_SIZE.MICRO && CONFIG.IS_HYBRID_APP}
-                pressOnEnter
-            />
-        </>
+        if (shouldCreateWorkspace) {
+            setOnboardingAdminsChatReportID(adminsChatReportID);
+            setOnboardingPolicyID(policyID);
+        }
+
+        completeOnboarding({
```

Comment:
i dont think this is a bug in your PR but while locally testing i noticed the completeGuidedSetup API call from here is not called when we redirect to oldDot. Do you know what changes will need to be made to resolve that and is that something you can easily add to your PR. I want to make sure completeGuidedSetup is always called before we redirect to oldDot. Let me know if its something easy to fix here, if not i can create another issue.